cpp/src/sampling/detail/sampling_utils_impl.cuh

```diff
@@ -285,5 +302,283 @@ sample_edges(raft::handle_t const& handle,
+        cugraph::per_v_random_select_transform_outgoing_e(
+          handle,
+          graph_view,
+          vertex_frontier.bucket(0),
+          edge_src_dummy_property_t{}.view(),
+          edge_dst_dummy_property_t{}.view(),
+          *edge_weight_view,
+          sample_edges_op_t<vertex_t>{},
+          rng_state,
+          fanout,
+          with_replacement,
+          std::make_optional<thrust::tuple<vertex_t, vertex_t, weight_t>>(
+            cugraph::invalid_vertex_id<vertex_t>::value,
+            cugraph::invalid_vertex_id<vertex_t>::value,
+            weight_t{}),
+          true);
+
+      std::tie(majors, minors, weights) = std::move(sample_e_op_results);
+    }
+  } else {
+    if (edge_id_type_view) {
+      auto [sample_offsets, sample_e_op_results] =
+        cugraph::per_v_random_select_transform_outgoing_e(
+          handle,
+          graph_view,
+          vertex_frontier.bucket(0),
+          edge_src_dummy_property_t{}.view(),
+          edge_dst_dummy_property_t{}.view(),
+          *edge_id_type_view,
+          sample_edges_op_t<vertex_t>{},
+          rng_state,
+          fanout,
+          with_replacement,
+          std::make_optional(thrust::make_tuple(cugraph::invalid_vertex_id<vertex_t>::value,
+                                                cugraph::invalid_vertex_id<vertex_t>::value,
+                                                edge_t{},
+                                                edge_type_t{})),
+          true);
+      std::tie(majors, minors, edge_ids, edge_types) = std::move(sample_e_op_results);
+    } else {
+      auto [sample_offsets, sample_e_op_results] =
+        cugraph::per_v_random_select_transform_outgoing_e(
+          handle,
+          graph_view,
+          vertex_frontier.bucket(0),
+          edge_src_dummy_property_t{}.view(),
+          edge_dst_dummy_property_t{}.view(),
+          edge_dummy_property_t{}.view(),
+          sample_edges_op_t<vertex_t>{},
+          rng_state,
+          fanout,
+          with_replacement,
+          std::make_optional<thrust::tuple<vertex_t, vertex_t>>(
+            cugraph::invalid_vertex_id<vertex_t>::value,
+            cugraph::invalid_vertex_id<vertex_t>::value),
+          true);
+      std::tie(majors, minors) = std::move(sample_e_op_results);
+    }
+  }
+
+  if (weights) {
+    auto edge_first = thrust::make_zip_iterator(
+      thrust::make_tuple(majors.begin(), minors.begin(), (*weights).begin()));
+    auto end_iter =
+      thrust::remove_if(handle.get_thrust_policy(),
+                        edge_first,
+                        edge_first + majors.size(),
+                        [] __device__(auto tuple) {
+                          auto v1 = thrust::get<0>(tuple);
+                          auto v2 = thrust::get<1>(tuple);
+
+                          return ((v1 == cugraph::invalid_vertex_id<vertex_t>::value) ||
+                                  (v1 == cugraph::invalid_vertex_id<vertex_t>::value));
```

Comment:
Shouldn't this be `v2 == cugraph::invalid_vertex_id<vertex_t>::value`?

Reply:
And this `remove_if` is unnecessary if we don't provide an invalid value when calling the above primitive.